src/config.ts

```diff
@@ -10,4 +10,5 @@ export const DEFAULT_CONFIG: SWACLIConfig = {
   appArtifactLocation: `.${path.sep}`,
   appBuildCommand: "npm run build --if-present",
   apiBuildCommand: "npm run build --if-present",
+  swaConfigFilePattern: /(routes|staticwebappsconfig)\.json/,
```

Comment:
The routes.json traversal algorithm leads to some problems where the wrong routes.json is chosen. E.g. Nuxt holds the internal nuxt-router config in a file called router.json inside of the .nuxt folder. 
I have a monorepo with an `api` folder and a `frontend` folder. When I specify the frontend folder as app-artifact-location, it will always evaluate to the routes.json in the .nuxt folder. It will probably also iterate over the node_modules folder, where there is a high chance of a routes,json to be existent. 

![image](https://user-images.githubusercontent.com/15030951/108439145-0105b200-7251-11eb-9d39-652ce7f973d1.png)

I would recommend to avoid recursion and just evaluate the parent level of the app-artifact-location or maybe introduce a routes_location variable like in the Github workflow.

Reply:
Thank you for reporting this behavior! I will add more check to the traversal algorithm.